lib/util/src/logging.c

```diff
@@ -1007,8 +1022,110 @@ package_source_info(struct json_context *jc)
     json_element_field(jc, "hse_version", "%s", HSE_VERSION_STRING);
 }
 
+static bool
+slog_tab_process_fmt(const char *fmt, char *new_fmt, size_t size)
+{
+    char *s = strchr(fmt, '%');
+
+    if (!s)
+        return false;
+
+    switch (*(++s)) {
+        case 's':
+        case 'p':
+        case 'x':
+        case 'X':
+            snprintf(new_fmt, size, "%s ", fmt);
+            return true;
+    }
+
+    snprintf(new_fmt, size, "%s ", fmt);
```

Comment:
is it possible to know if this is the last field so you don't append the extra space?

Reply:
Don't indent case labels.
Is this function just eating format specifiers?